libmachine/log/log.go

```diff
@@ -1,124 +1,76 @@
 package log
 
-import (
-	"io"
-	"os"
-	"sync"
-)
+var logger Logger
 
-// Logger - Why the interface?  We may only want to print to STDOUT and STDERR for now,
-// but it won't neccessarily be that way forever.  This interface is intended
-// to provide a "framework" for a variety of different logging types in the
-// future (log to file, log to logstash, etc.) There could be a driver model
-// similar to what is done with OS or machine providers.
-type Logger interface {
-	Debug(...interface{})
-	Debugf(string, ...interface{})
-
-	Error(...interface{})
-	Errorf(string, ...interface{})
-
-	Info(...interface{})
-	Infof(string, ...interface{})
-
-	Fatal(...interface{})
-	Fatalf(string, ...interface{})
-
-	Print(...interface{})
-	Printf(string, ...interface{})
-
-	Warn(...interface{})
-	Warnf(string, ...interface{})
-
-	WithFields(Fields) Logger
+// SetLogger sets the logger for libmachine, must implement the Logger interface
+func SetLogger(lgr Logger) {
+	logger = lgr
 }
 
-var (
-	l = StandardLogger{
-		mu: &sync.Mutex{},
+// GetLogger gets the currently set logger for libmachine
+func GetLogger() Logger {
+	if logger == nil {
+		logger = newDefaultLogger()
 	}
-	IsDebug = false
-)
-
-type Fields map[string]interface{}
-
-func init() {
-	// TODO: Is this really the best approach?  I worry that it will create
-	// implicit behavior which may be problmatic for users of the lib.
-	SetOutWriter(os.Stdout)
-	SetErrWriter(os.Stderr)
-}
-
-func SetOutWriter(w io.Writer) {
-	l.OutWriter = w
-}
-
-func SetErrWriter(w io.Writer) {
-	l.ErrWriter = w
+	return logger
 }
 
+// Debug logs a message at level Debug
 func Debug(args ...interface{}) {
-	l.Debug(args...)
+	GetLogger().Debug(args...)
```

Comment:
Why the whole `GetLogger` song and dance?  Why not just initialize `logger` in a `var` block in the top of the module? e.g.

```go
var (
    logger = newDefaultLogger()
)
```